google/cloud/sql/connector/instance.py

```diff
@@ -326,11 +327,11 @@ async def _perform_refresh(self) -> InstanceMetadata:
             try:
                 metadata = await metadata_task
                 # check if automatic IAM database authn is supported for database engine
-                if self._enable_iam_auth and not metadata[
-                    "database_version"
-                ].startswith("POSTGRES"):
+                if self._enable_iam_auth and metadata["database_version"].startswith(
+                    "SQLSERVER"
```

Comment:
This worries me because it doesn't capture any future database versions that might be launched that don't start with SQLServer

Reply:
Good catch! Updated logic to be `not metadata["database_version"].startswith(("POSTGRES", "MYSQL"))` to account for this.